src/BuildCheck.UnitTests/EndToEndTests.cs

```diff
@@ -236,9 +235,12 @@ public void NoEnvironmentVariableProperty_Test(bool? customConfigEnabled, string
             customConfigData);
 
         string output = RunnerUtilities.ExecBootstrapedMSBuild(
-            $"{Path.GetFileName(projectFile.Path)} /m:1 -nr:False -restore -check", out bool success, false, _env.Output);
+            $"{Path.GetFileName(projectFile.Path)} /m:1 -nr:False -restore -check", out bool success, false, _env.Output, timeoutMilliseconds: 12000_000);
```

Comment:
From one test the timeout was removed and in this one it was added, which one are we going with? Yes timeout, no timeout?

Reply:
No timeout is preferable (the utils will fallback to default 30 seconds timeout).
Historicaly some of the unittests were adding longer timeouts as we had a perf issue and it was an attempt to workaround - this is not legitimate reason anymore.
Sometimes (this is the case here likely) - devs are adding high timeouts to be able to debug bootstrapped msbuild, without runner utils killing it after timeout. The more prefered way is setting the `MSBuildDebugUnitTests` env var - which then stops enforcing any timeout at all. The env var can be set as a global variable on dev machine - so one doesnt have to care about it anymore